x-pack/legacy/plugins/siem/public/components/embeddables/embedded_map.tsx

```diff
@@ -53,6 +55,8 @@ export const EmbeddedMap = React.memo<EmbeddedMapProps>(
     const [loadingKibanaIndexPatterns, kibanaIndexPatterns] = useIndexPatterns();
     const [siemDefaultIndices] = useKibanaUiSetting(DEFAULT_INDEX_KEY);
 
+    const portalNode = React.useMemo(() => createPortalNode(), []);
```

Comment:
`createPortalNode()` feels like an effect, which makes me wonder if `useEffect` is technically more appropriate than `useMemo` here, due to following statement [in the useMemo reference](https://reactjs.org/docs/hooks-reference.html#usememo):

> Remember that the function passed to useMemo runs during rendering. Don’t do anything there that you wouldn’t normally do while rendering. For example, side effects belong in useEffect, not useMemo.

What do you think?

Reply:
I was following suit from the [example docs](https://github.com/httptoolkit/react-reverse-portal/blob/master/README.md#getting-started), but as you pointed out [later in the docs](https://github.com/httptoolkit/react-reverse-portal/blob/master/README.md#portalscreateportalnode) it certainly sounds like an effect. Feels like a grey area -- let's keep an eye on any wonky behavior, but maintain it as described in the example. 

Thanks for calling attention to this 🙂